vehicles/src/common/helper/validate-loa.helper.ts

```diff
@@ -0,0 +1,210 @@
+import { ReadLoaDto } from 'src/modules/special-auth/dto/response/read-loa.dto';
+import { PermitType } from '../enum/permit-type.enum';
+import { Loas, PermitData } from '../interface/permit.template.interface';
+import { Permit } from 'src/modules/permit-application-payment/permit/entities/permit.entity';
+import * as dayjs from 'dayjs';
+import { In, QueryRunner } from 'typeorm';
+import { LoaDetail } from 'src/modules/special-auth/entities/loa-detail.entity';
+import { Mapper } from '@automapper/core';
+import { UnprocessableEntityException } from '@nestjs/common';
+
+export const isVehicleTypeValid = (
+  permitVehicleType: string,
+  permitVehicleId: string,
+  powerUnits?: string[],
+  trailers?: string[],
+): boolean => {
+  const isPowerUnitAllowed =
+    permitVehicleType === 'powerUnit'
+      ? powerUnits.includes(permitVehicleId)
+      : true;
+
+  const isTrailerAllowed =
+    permitVehicleType === 'trailer' ? trailers.includes(permitVehicleId) : true;
+
+  return isPowerUnitAllowed && isTrailerAllowed;
+};
+
+export const isPermitTypeValid = (
+  permitTypePermit: PermitType,
+  permitType: PermitType[],
+): boolean => {
+  return permitType.includes(permitTypePermit);
+};
+
+export const isValidDateForLoa = (
+  loaDetail: Loas | ReadLoaDto,
+  permit: Permit,
+): boolean => {
+  const { startDate, expiryDate } = loaDetail;
+  const { startDate: permitStartDate, expiryDate: permitExpiryDate } =
+    permit.permitData;
+  return (
+    isStartDateValid(startDate, permitStartDate) &&
+    isEndDateValid(expiryDate, permitExpiryDate)
+  );
+};
+
+export const isStartDateValid = (
+  startDate: string,
+  permitStartDate: string,
+): boolean => {
+  return (
+    dayjs(startDate).isBefore(permitStartDate, 'day') ||
+    dayjs(startDate).isSame(permitStartDate, 'day')
+  );
+};
+
+export const isEndDateValid = (
+  expiryDate: string,
+  permitExpiryDate: string,
+): boolean => {
+  return expiryDate
+    ? dayjs(expiryDate).isAfter(permitExpiryDate, 'day') ||
+        dayjs(expiryDate).isSame(permitExpiryDate, 'day')
+    : true;
+};
+export const isValidLoa = async (
+  permit: Permit,
+  queryRunner: QueryRunner,
+  mapper: Mapper,
+) => {
+  const { companyId } = permit.company;
+  const permitData = JSON.parse(permit.permitData.permitData) as PermitData;
+  const { vehicleId: permitVehicleId, vehicleType: permitVehicleType } =
+    permitData.vehicleDetails;
+  if (permitData.loas) {
+    const loaNumbers = permitData.loas.map((loa) => loa.loaNumber);
+    const readLoaDto = await findLoas(
+      companyId,
+      loaNumbers,
+      queryRunner,
+      mapper,
+    );
+
+    // Validate LOA details and permit data against database entries
+    validateLoaDetails(readLoaDto, permit, permitVehicleId, permitVehicleType);
+
+    // validate LoA snapshot in permit Data
+    validatePermitDataAgainstLoas(
```

Comment:
What's the difference between these two functions? The both seemingly have the same set of checks on the same data.